### ISSUE TITLE
NLog.config is not read when application is stored in a drive's root folder

### ISSUE THREAD

**Issue by wilhelmmedetz** [NLog/NLog#4314]
**NLog version**: 4.7.7
**Platform**: .Net 4.5

- What is the current result?
  - NLog.config is not read
- What is the expected result?
  - NLog.config should be read
- Did you checked the [Internal log]
  - not applicable
- Please post full exception details (message, stacktrace, inner exceptions)
  - none 
- Are there any workarounds? yes/no
  - `applicationname.exe.nlog` or `NLog.dll.nlog` works
- Is there a version in which it did work? 
  - NLog.4.2.3
- Can you help us by writing an unit test?
  - no

When starting an application from a drive's root directory the NLog.config in the root directory is not read.

Steps to reproduce:
- create a simple console application with a NLog.config
- map the build directory with subst to some drive
   e.g. `subst h: C:\...\ConsoleApp\bin\Debug\net452`
- start the console application on the mapped drive
- NLog.config is not read

**Comment by snakefoot** [comment]
I can confirm that it also fails on my machine:

```
2021-02-24 20:27:21.8316 Debug Loading config from D:NLog.config
2021-02-24 20:27:21.8316 Error Failed loading from config file location: D:NLog.config Exception: System.UriFormatException: Invalid URI: A Dos path must be rooted, for example, 'c:\'.
   at System.Uri.CreateThis(String uri, Boolean dontEscape, UriKind uriKind)
   at System.Xml.XmlResolver.ResolveUri(Uri baseUri, String relativeUri)
   at System.Xml.XmlTextReaderImpl..ctor(String uriStr, XmlReaderSettings settings, XmlParserContext context, XmlResolver uriResolver)
   at System.Xml.XmlReaderSettings.CreateReader(String inputUri, XmlParserContext inputContext)
   at NLog.Internal.Fakeables.AppEnvironmentWrapper.LoadXmlFile(String path)
   at NLog.Config.LoggingConfigurationFileLoader.LoadXmlLoggingConfigurationFile(LogFactory logFactory, String configFile)
   at NLog.Config.LoggingConfigurationFileLoader.TryLoadLoggingConfiguration(LogFactory logFactory, String configFile, LoggingConfiguration& config)
2021-02-24 20:27:21.8650 Debug Found assembly location directory: 'D:\' (ConsoleApp2, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null)
2021-02-24 20:27:21.8650 Debug Targets not configured for logger: ConsoleApp2.Program

```

**Comment by snakefoot** [comment]
Think bug was introduced with #2846 in NLog 4.5.9

**Comment by snakefoot** [comment]
@wilhelmmedetz Thank you for reporting this issue. Have created #4316 to fix the bug.

**Comment by snakefoot** [comment]
Resolved by #4316

**Comment by snakefoot** [comment]
NLog ver. 4.7.8 has been released: https://www.nuget.org/packages/NLog
